src/server/tests/server_test.go

```diff
@@ -109,20 +105,24 @@ func TestProtocols(t *testing.T) {
 		srv, cli, close := test_utils.StartServerAndConnect("gopher123", "mytotallyrealpassword", "/register")
 		defer close()
 
-		populate(srv, 51)
-		test_utils.MsgHistory(t, cli, 0, time.Now().UnixNano())
+		test_utils.Populate(srv, 51)
+		mock := mockclient.MockClient{}
+		mock.MsgHistory(t, cli, 0, time.Now().UnixNano())
 
-		got := len(test_utils.RecvMsgHistory(t, cli).Messages)
+		got := len(mock.RecvMsgHistory(t, cli).Messages)
 		expected := 50
 		test_utils.AssertEqual(t, got, expected)
 	})
+}
 
+func TestErrorHandling(t *testing.T) {
 	t.Run("database error", func(t *testing.T) {
 		srv, cli, close := test_utils.StartServerAndConnect("gopher123", "pass", "/register")
 		defer close()
 
-		srv.Database.ShouldErrorOnNext = true
-		test_utils.MsgSend(t, cli, "this should error.")
+		srv.Database.(*database.InMemoryDatabase).ShouldErrorOnNext = true
```

Comment:
Violation of encapsulation. This should be a setter function. Note that the flag needs synchronization to ensure it can safely be called from concurrent goroutines.

Reply:
Looking at it again, `Database` shouldn't be directly accessed here either.

Reply:
Why should the database not be directly acessed?  I belive that is the best way to accomplish this.

Reply:
Discussed on Discord. Will be handled in a separate issue.